Block/AccountBlockService.php

```diff
@@ -31,22 +30,22 @@
 class AccountBlockService extends BaseBlockService
 {
     /**
-     * @var ProfileMenuBuilder
+     * @var TokenStorageInterface
      */
-    private $securityContext;
+    private $tokenStorage;
 
     /**
      * Constructor.
      *
-     * @param string                   $name
-     * @param EngineInterface          $templating
-     * @param SecurityContextInterface $securityContext
+     * @param string                $name
+     * @param EngineInterface       $templating
+     * @param TokenStorageInterface $tokenStorage
      */
-    public function __construct($name, EngineInterface $templating, SecurityContextInterface $securityContext)
+    public function __construct($name, EngineInterface $templating, TokenStorageInterface $tokenStorage)
```

Comment:
Not sure it is that simple… this interface does not exist in old versions of symfony that we still support… [it was introduced in sf 2.6](https://github.com/symfony/symfony/commit/b96778718507e520596279954937f75c181efc17).

What you can do is remove the type hinting, and inject either the old thing and the new thing, depending on the version of the security component, and add `NEXT_MAJOR` markers explaining what to do when dropping compatibility with old versions of symfony